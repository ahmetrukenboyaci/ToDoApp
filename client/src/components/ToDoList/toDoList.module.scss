.toDoListContainer {
  background-color: white;
  overflow: auto;
  height: calc(100% - 147px);
}