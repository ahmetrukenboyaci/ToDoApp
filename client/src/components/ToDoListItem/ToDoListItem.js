/** Dependencies **/
import React, {useCallback} from 'react';
import PropTypes from 'prop-types';

/** Styles **/
import styles from './toDoListItem.module.scss';

/** Images **/
import Uncompleted from '../../assets/images/Uncompleted';
import Completed from '../../assets/images/Completed';
import Delete from '../../assets/images/Delete';

const ToDoListItem = ({data, onDelete, onComplete}) => {
	const onToDoDelete = useCallback(() => {
		onDelete(data._id);
	}, [data?._id, onDelete]);

	const onToDoComplete = useCallback(() => {
		onComplete(data._id, data.done);
	}, [data?._id, data?.done, onComplete]);

	return (
		<div id={'toDoListItemContainer'} className={`${styles.toDoListItemContainer} ${data?.done ? styles.done : ''}`}>
			<div id={'text'} className={styles.text}>
				{data?.description}
			</div>
			<div className={styles.iconContainer}>
				<div onClick={onToDoComplete} className={styles.toggleCompletion}>
					{data?.done ? <Completed height={20}/> : <Uncompleted height={20}/>}
				</div>
				<div onClick={onToDoDelete} className={styles.deleteToDo}>
					<Delete height={20}/>
				</div>
			</div>
		</div>
	);
};

ToDoListItem.propTypes = {
	data: PropTypes.object,
	onDelete: PropTypes.func,
	onComplete: PropTypes.func
};

export default ToDoListItem;