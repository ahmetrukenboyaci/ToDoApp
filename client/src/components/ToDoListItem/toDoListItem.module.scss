.toDoListItemContainer {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 16px;
  margin: 8px;
  border: 1px solid #eee;
  border-radius: 4px;

  &.done {
    background-color: #ddd;
  }

  & .text {

  }

  & .iconContainer {
    display: flex;
    justify-content: center;
    align-items: center;

    & .toggleCompletion {
      display: flex;
      margin-right: 5px;
      cursor: pointer;
    }

    & .deleteToDo {
      display: flex;
      margin-left: 5px;
      cursor: pointer;
    }
  }
}