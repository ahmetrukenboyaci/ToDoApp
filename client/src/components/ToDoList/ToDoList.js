/** Dependencies **/
import React from 'react';
import PropTypes from 'prop-types';

/** Components **/
import ToDoListItem from '../ToDoListItem/ToDoListItem';

/** Styles **/
import styles from './toDoList.module.scss';

const ToDoList = ({data, onDelete, onComplete}) => {
	return (
		<div className={styles.toDoListContainer}>
			{data.map((item) => {
				return <ToDoListItem key={item._id} data={item} onDelete={onDelete} onComplete={onComplete}/>;
			})}
		</div>
	);
};

ToDoList.propTypes = {
	data: PropTypes.array,
	onDelete: PropTypes.func,
	onComplete: PropTypes.func
};

ToDoList.defaultProps = {
	data: []
};

export default ToDoList;