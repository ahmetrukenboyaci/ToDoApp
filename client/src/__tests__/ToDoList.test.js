import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';
import ToDoList from '../components/ToDoList/ToDoList';
import ToDoListItem from '../components/ToDoListItem/ToDoListItem';

configure({adapter: new Adapter()});

describe('ToDoList', () => {
	it('should render todolist component', () => {
		// eslint-disable-next-line no-unused-vars
		const wrapper = shallow(<ToDoList/>);
	});

	it('should render initial layout', () => {

		const component = shallow(<ToDoList/>);

		expect(component.getElements()).toMatchSnapshot();
	});

	it('should render todoItem', () => {
		const data = [
			{ _id: '1', description: 'a', done: false},
			{ _id: '2', description: 'b', done: false}
		];

		const component = shallow(<ToDoList data={data}/>);

		expect(component.find(ToDoListItem)).toHaveLength(data.length);
	});
});