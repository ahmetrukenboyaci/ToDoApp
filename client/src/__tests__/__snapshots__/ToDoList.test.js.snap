// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList should render initial layout 1`] = `
Array [
  <div
    className="toDoListContainer"
  />,
]
`;
